### ISSUE TITLE
Home buffer at full screen not centered

### ISSUE THREAD

**Issue by don-han** [syl20bnr/spacemacs#5120]
#### Description
The home buffer is not centered when window is maximized. I tested it both on `master` and on `develop`

![image](https://cloud.githubusercontent.com/assets/5600864/13136906/44db2490-d5d2-11e5-8f2d-ad8ba471e9af.png)

#### Expected Behavior
Not sure if this is a feature, but I expected the home buffer to be on the center

#### System Info
- OS: darwin
- Emacs: 24.5.1
- Spacemacs: 0.105.9
- Spacemacs branch: master (rev. 0562f05)
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
((auto-completion :variables auto-completion-return-key-behavior 'complete auto-completion-tab-key-behavior 'cycle auto-completion-complete-with-key-sequence nil)
 better-defaults emacs-lisp markdown syntax-checking version-control
 (latex :variables latex-enable-auto-fill t)
 themes-megapack python)
```

**Comment by StreakyCobra** [comment]
When https://github.com/syl20bnr/spacemacs/pull/4262 will be merged it will be easy to do probably.

**Comment by TheBB** [comment]
I'm leaning towards wontfix for this. I'm not sure starting up in a distraction-free mode would be very useful.

**Comment by StreakyCobra** [comment]
To be clear, from my point of view this should not be the default, but it should be easy to do at user-config level once a distraction-free mode would be merged.

**Comment by syl20bnr** [comment]
Same here, it would feel weird for a lot of users.

**Comment by a13ph** [comment]
Centering might be not the preferred solution - though I fail to see how, tbh, you don't need to activate distraction-free modes for that - but having lots of unused space with left alignment does suck.

**Comment by StreakyCobra** [comment]
I was just saying this would be possible *easily* at user level with a distraction-free mode, not that this should be the Spacemacs solution. If people want to create a PR that does this, please feel free. Also be sure to deal with the arbitrary length of the recent files list.

**Comment by StreakyCobra** [comment]
This has been fixed in `develop` by #6048

**Comment by mosic** [comment]
How can I switch back to the old layout? Centered does look better on small screens, but it's pretty bad on larger ones (especially with release notes visible):

![image](https://cloud.githubusercontent.com/assets/1229422/15291466/6424bc84-1b7e-11e6-8554-080e27077c1a.png)

**Comment by ralesi** [comment]
Are we just looking to change the width of the release notes?

**Comment by StreakyCobra** [comment]
@syl20bnr :point_up: :point_up: told you :-)

**Comment by kRITZCREEK** [comment]
![](https://imgs.xkcd.com/comics/workflow.png)

I'll just leave this here...

**Comment by mosic** [comment]
:) 

It's not just that the layout is changed, it's not exactly centered. You can see in the screenshot that there is a noticeable gap on the right side when Emacs is started, which causes the Spacemacs logo to be off-center. If I resize to smaller size than back to maximized and press `C-r`, the full width is being used and then it looks fine.

**Comment by mosic** [comment]
@ralesi It seems reasonable to me to limit release notes width to something like 80-100 characters.

**Comment by syl20bnr** [comment]
:smiley: 

It is not really centered here too, it depends on timing when starting Emacs.
If people don't like it then let's make it an option and turn off by default.
I actually like the release notes taking all the page.

What if we center the release note as well and draw a separation line with the other info that will not be centered ? (like 275 packages loaded...)

**Comment by syl20bnr** [comment]
Or... we can totally revamp the page too, proposals are welcome  :-D

**Comment by izahn** [comment]
I actually really like the centered logo and such. But it doesn't update when the window size changes. Can we do something like 

```
  ;; Recenter spacemacs logo on window change
  (defun my-spacemacs-buffer/refresh ()
    "spacemacs-buffer/refresh with dummy argument"
    (interactive)
    (with-selected-window (get-buffer-window "*spacemacs*")
      (spacemacs-buffer/refresh)))

  (add-to-list 'window-size-change-functions 'my-spacemacs-buffer/refresh)
```
so that it stays centered?

Actually `window-size-change-functions` is only a partial solution since it doesn't get triggered when re-sizing the frame using the window manager. But I think something like this is better than nothing.

**Comment by MadAnd** [comment]
On my 12'' laptop home buffer content is too wide upon startup:
![screenshot_20160522_141642](https://cloud.githubusercontent.com/assets/2280844/15453763/0949f792-202a-11e6-893b-c0e7a7a184bc.png)

It is due to [these lines](https://github.com/syl20bnr/spacemacs/blob/f305bd01f95326b089f1c87135e0ffea0756c15f/core/core-spacemacs-buffer.el#L787-788) being evaluated before `dotspacemacs-default-font` settings went into effect, and thus `(window-total-width)` returns wrong value at that moment. I have no idea what is the proper way to fix this though.

Related question. If I later press `C-r` (`spacemacs-buffer/refresh`), the buffer is being re-centered correctly, but the Changelog block disappears. Is it a bug or by design?

**Comment by syl20bnr** [comment]
The framed text should still be here after a C-r so this looks like a bug.

**Comment by ralesi** [comment]
I have been looking at ways to get this automatically centered.  It looks like `window-size-change-functions` would work, but it doesn't seem to run when I am resizing the frame.  `window-configuration-change-hook` works, but it is a bit overkill, and it also only returns the selected window, so modification and deletion of the spacemacs-buffer fires off another change while running.    

I think the issues with the buffer not completely centered are tied to various fringe sizes.  I played with a few sizes and noticed truncation on the spacemacs version number.

**Comment by syl20bnr** [comment]
Nope we don't have this. We only provide a way to maximize the frame at the end of init.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
